File: Chapter 05/JS/section_5_4_server_monitor/before/serverMonitor.js
/**
 * The Core Business Logic.
 * WARNING: This class is a "Liability" because it violates the 
 * Golden Rule of Separation of Concerns.
 */
class ServerMonitor {
    /**
     * Checks the server temperature and sends an alert if it's too high.
     * @param {number} temp - The current temperature reading.
     */
    checkTemperature(temp) {
        // VIOLATION: Hardcoded "magic number". 
        // This should be a configurable threshold.
        if (temp > 95) {
            // VIOLATION: Tight Coupling.
            // We are 'new-ing' up a concrete dependency inside our logic.
            // This makes the class impossible to unit test without a live API.
            const twilio = new TwilioClient("API_KEY");
            twilio.sendSms("555-1234", "Server is overheating!");
        } else {
            console.log(`Temp ${temp} is nominal.`);
        }
    }
}

/**
 * Mock of a 3rd party SMS library.
 * In a real system, this is the "Chaotic Outside World".
 */
class TwilioClient {
    constructor(key) {
        this.key = key;
    }
    
    sendSms(to, body) {
        console.log(`[Twilio API] Sending SMS to ${to}: ${body}`);
    }
}

module.exports = ServerMonitor;

File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/before/ServerMonitor.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.before;

/**
 * The Core Business Logic.
 * This class currently fails as a "Boundary Keeper" because it allows
 * external infrastructure details to leak into the domain.
 */
public class ServerMonitor {
    
    /**
     * Evaluates server health based on temperature.
     * @param temp The temperature value to check.
     */
    public void checkTemperature(int temp) {
        // VIOLATION: Hardcoded magic number.
        if (temp > 95) {
            // VIOLATION: Direct Dependency.
            // By hardcoding the 'TwilioClient', we have abandoned our post
            // as a Clarity Engineer.
            TwilioClient twilio = new TwilioClient("API_KEY");
            twilio.sendSms("555-1234", "Server is overheating!");
        } else {
            System.out.println("Temp " + temp + " is nominal.");
        }
    }
}

/**
 * Simulates a third-party Library.
 */
class TwilioClient {
    public TwilioClient(String key) { }
    public void sendSms(String to, String body) {
        System.out.println("[Twilio API] Sending SMS to " + to + ": " + body);
    }
}